### ISSUE TITLE
Dependency on safe-buffer breaks browser builds

### ISSUE THREAD

**Issue by BobbyRadford** [near/wallet-selector#756]
**Describe the bug**
The `@near-wallet-selector/core` dependency on `near-api-js` and `safe-buffer` breaks builds for the browser.

Output errors such as this:
```sh
✘ [ERROR] Could not resolve "buffer"

    node_modules/safe-buffer/index.js:2:21:
      2 │ var buffer = require('buffer')
        ╵                      ~~~~~~~~

  The package "buffer" wasn't found on the file system but is built into node.
  Are you trying to bundle for node? You can use "platform: 'node'" to do
  that, which will remove this error.
```

Not trying to build for Node here, so that doesn't help.

**To Reproduce**
Using a tool like esbuild or estrella, set platform to "browser" (default value)
```javascript
await build({
  bundle: true,
  entry: `${__dirname  }/src/scripts/index.tsx`,
  outdir: `${__dirname }/dist/static/scripts`,
  minify: true,
  format: 'esm',
  target: 'es2017',
  platform: 'browser'
});
```

**Expected behavior**
Since near-wallet-selector is meant for browser web apps, it shouldn't have dependency on Node libraries. Builds for the browser should work out-of-the-box without any workarounds.

**Additional context**
I'm also happy to hear any suggestions on how to build for the browser today.

**Comment by kujtimprenkuSQA** [comment]
Hi, @BobbyRadford thank you for raising this issue.

`Buffer` is not used directly by `@near-wallet-selector/core` but it's used by `near-api-js`.

In the modern bundlers node native libraries are not added in the browser-based projects by default so you will need to provide `buffer` as a polyfill. 

Have you tried installing `buffer` as a dependency or a devDependency? ( in `esbuild` the error goes away if you install it.)

**Comment by kujtimprenkuSQA** [comment]
Yeah, it makes sense to use only "browser-compatible" libraries/modules but at the moment `buffer` is used by `near-api-js` a lot.

From my investigation so far I found out that `buffer` and `process` are being used by `near-api-js` so you must provide them as polyfills in your dApp.

Also just installing the `buffer` or `process` package in a dApp created with `esbuild` does not completely fix the issue, i.e installing the `@near-wallet-selector/my-near-wallet` and trying to sign in from the modal would show an error: 
**"Buffer is not defined"**

Since I am not very familiar with `esbuild` the easiest way I could do to fix this issue was by:

1. Install `buffer` and `process` as dependencies

2. Create a `polyfills.ts` in your project and add: 
```ts
import * as process from "process";
import { Buffer } from "buffer";
Object.assign(self, {
  process,
  global: self,
  Buffer,
});
```

3. Import `polyfills.ts` in your main ts/tsx file.

This is a simple demo (copy of wallet-selector/examples/react with esbuild):
https://github.com/kujtimprenkuSQA/guest-book-esbuild

This is the commit where I added `buffer` and `process` as polyfills:
https://github.com/kujtimprenkuSQA/guest-book-esbuild/commit/98928f9b31b5f27815b6c13ccb33a870825143fb

I am not very familiar with this bundler though so forgive my lack of  following the "best practices" in this case.
